src/components/MenuBar/MenuBar.tsx

```diff
@@ -66,7 +66,10 @@ export default function MenuBar() {
 
   const handleSubmit = (event: FormEvent<HTMLFormElement>) => {
     event.preventDefault();
-    window.history.replaceState(null, '', window.encodeURI(`/room/${roomName}`));
+    // If this app is deployed as a twilio function, don't change the URL beacuse routing isn't supported.
+    if (!window.location.origin.includes('twil.io')) {
```

Comment:
This is ok for now. But not fool proof since domain can change. Another way (potentially better) is to detect if you're running in a twilio function environment then set a variable before returning the page. Kinda how server-side rendering engine works. I'm ok to punt it if you think that takes time to implement.

Reply:
I'll think about solutions for this, but I don't think we can use a server-side rendering solution. There's no server under our control that is serving the static web assets.  It's kinda like viewing a site that is hosted from an S3 bucket.

A build-time environment variable could be an option, but that has the risk that maybe the user will not set it correctly. I'll try to think of a better solution here, but for now I'm leaning towards punting it 🏈 .

Reply:
Ok sounds good.